Seed flaky test `TestSamplePPC.test_normal_scalar`

mattiadg opened the pull request:
This is related to the issue 
* Remove call to function with unused output
* Add assert to test function with list input
* Add comment to refer to the issue

<!-- !! Thank your for opening a PR !! -->

**What is this PR about?**
The issue Flaky test: test_normal_scalar #6211 reported a Flaky test in test_normal_scalar https://github.com/pymc-devs/pymc/blob/main/pymc/tests/test_sampling.py#L617. I could not reproduce it yet, but I found other minor issues in the same test: calls to the tested function that were not calling any assert. I added an assert after the line with an input different from the other calls, and removed the other line that looks redundant. Under the suggestion of @michaelosthege, I added a comment asking to report further failing of this test under the same issue

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- Seed flaky test `TestSamplePPC.test_normal_scalar`

mattiadg committed:
Partially fix test_normal_scalar

This is related to the issue Flaky test: test_normal_scalar #6211 https://github.com/pymc-devs/pymc/issues/6211

* Remove call to function with unused output
* Add assert to test function with list input
* Add comment to refer to the issue

mattiadg commented:
Apparently, that call was covering some additional lines but no assert was there. I will understand what it was supposed to test and add an assert accordingly

michaelosthege commented:
> Apparently, that call was covering some additional lines but no assert was there. I will understand what it was supposed to test and add an assert accordingly

Not sure what you mean. For me this looks all good already.

michaelosthege commented:
Relevant comment: https://github.com/pymc-devs/pymc/issues/6210#issuecomment-1280049067

Looks like the `SeededTest` does not affect posterior predictive sampling?!

mattiadg commented:
Found the problem, we need to pass explicitly the `random_seed`, actually an object of type RandomState to `pm.sample_posterior_predictive`. This way I get the same samples. New commit follows in a while

mattiadg committed:
Fix test_normal_scalar

This really makes the samples fixed and reproducible. There are functions called by sample_posterior_predictive that ignore the global random seed if one is not provided explicitely.

mattiadg committed:
Make test_component_choice_random deterministic to fix issue #6210

mattiadg committed:
Revert latest commit since there is another PR for that

ricardoV94 commented on pymc/tests/test_sampling.py:
```diff
@@ -625,23 +625,30 @@ def test_normal_scalar(self):
 
             # test default case
-            idata_ppc = pm.sample_posterior_predictive(trace, var_names=["a"])
+            idata_ppc = pm.sample_posterior_predictive(
+                trace, var_names=["a"], random_seed=random_state
+            )
```
Only this test requires a seed, the others are not checking anything that depends on the draws. Although passing a seed doesn't hurt, it makes reading the git history less straightforward (unless it was a commit about adding random_seed to all tests, and not just this one in particular).

mattiadg replied:
I see, removed the seed from all the others.

ricardoV94 reviewed (changes requested):
Looks great, just a request to minimize the PR changes

mattiadg committed:
Limit random_state to the only test that checks the content of a draw

ricardoV94 commented:
Thanks @mattiadg!
